modules/calib3d/include/opencv2/calib3d.hpp

```diff
@@ -1574,6 +1574,53 @@ CV_EXPORTS_W  int estimateAffine3D(InputArray src, InputArray dst,
                                    OutputArray out, OutputArray inliers,
                                    double ransacThreshold = 3, double confidence = 0.99);
 
+/** @brief Computes an optimal affine transformation between two 2D point sets.
+
+@param src First input 2D point set.
+@param dst Second input 2D point set.
+@param out Output 2D affine transformation matrix \f$2 \times 3\f$ .
+@param inliers Output vector indicating which points are inliers.
+@param ransacThreshold Maximum reprojection error in the RANSAC algorithm to consider a point as
+an inlier.
+@param confidence Confidence level, between 0 and 1, for the estimated transformation. Anything
+between 0.95 and 0.99 is usually good enough. Values too close to 1 can slow down the estimation
+significantly. Values lower than 0.8-0.9 can result in an incorrectly estimated transformation.
+
+The function estimates an optimal 2D affine transformation between two 2D point sets using the
+RANSAC algorithm.
+ */
+CV_EXPORTS_W int estimateAffine2D(InputArray src, InputArray dst,
+                                  OutputArray out, OutputArray inliers,
+                                  double ransacThreshold = 3, double confidence = 0.99);
+
+/** @brief Computes an optimal limited affine transformation with 4 degrees of freedom between
+two 2D point sets.
+
+@param src First input 2D point set.
+@param dst Second input 2D point set.
+@param out Output 2D affine transformation (4 degrees of freedom) matrix \f$2 \times 3\f$ .
+@param inliers Output vector indicating which points are inliers.
+@param ransacThreshold Maximum reprojection error in the RANSAC algorithm to consider a point as
+an inlier.
+@param confidence Confidence level, between 0 and 1, for the estimated transformation. Anything
+between 0.95 and 0.99 is usually good enough. Values too close to 1 can slow down the estimation
+significantly. Values lower than 0.8-0.9 can result in an incorrectly estimated transformation.
+
+The function estimates an optimal 2D affine transformation with 4 degrees of freedom limited to
+combinations of translation, rotation, and uniform scaling. Uses the RANSAC algorithm for robust
+estimation.
+
+Estimated transformation matrix is:
+\f[ \begin{bmatrix} \cos(\theta)s & -\sin(\theta)s & tx \\
+                \sin(\theta)s & \cos(\theta)s & ty
+\end{bmatrix} \f]
+Where \f$ \theta \f$ is the rotation angle, \f$ s \f$ the scaling factor and \f$ tx, ty \f$ are
+translations in \f$ x, y \f$ axes respectively.
+*/
+CV_EXPORTS_W int estimateAffinePartial2D(InputArray src, InputArray dst,
```

Comment:
Rather than an integer # of DOF, it might be better to take a "motionType" parameter the way that findTransformECC() does.  It takes MOTION_AFFINE (6 DOF) or MOTION_EUCLIDEAN (3 DOF).  We'd need to come up with a name for the 4 DOF case.  MOTION_TRANSLATE_ROTATE_SCALE?  Or MOTION_RIGID_PLUS_SCALE?  Or ???